examples/weather/diagnostic/config/diagnostic_precip.yaml

```diff
@@ -0,0 +1,68 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+model:
+  model_type: "afno"
+  model_name: "diagnostic-precip-afno"
+  inp_shape: [720, 1440]
+  in_channels: 27 # TODO: this could be automated
```

Comment:
You can use `eval` in yaml files for stuff like this. For example, https://github.com/loliverhennigh/modulus/blob/fea-unified_recipe_model_card/examples/weather/unified_recipe/conf/dataset/7_var.yaml#L25.

Reply:
I think it's a bit complex to do with eval because in addition to the number of variables in the data generator, `in_channels` also depends on how many auxiliaries/invariants you add. It's better to implement it using a function in the Python code that allows removing this input parameter.